test/ResetStaticStateExtension.groovy

```diff
@@ -27,6 +28,13 @@ public class ResetStaticStateExtension implements BeforeEachCallback,
 
     @Memoized
     public findAllResettableClasses() {
-        return new Reflections(Resettable.getPackage().getName()).getSubTypesOf( Resettable.class )
+        // There's an interesting problem here.
+        def resettableClasses = new Reflections(Resettable.getPackage().getName()).getSubTypesOf( Resettable.class )
+        def traits = resettableClasses.find { Traits.isTrait(it) }
+        traits.each {
+            resettableClasses += new Reflections(it.getPackage().getName()).getSubTypesOf( it )
+            resettableClasses.removeElement(it)
+        }
+        return resettableClasses
```

Comment:
I'm not sure about the correctness of this change? 

1. Find only finds a single result, even if there are multiple
2. the `traits.each` seems to be looking for subtypes - but when I inspect the resettableClasses initially, it already contains all the subtypes - so there's no need to re-add them.

I think this could be simplified, and fixed to handle multiple traits if that problem ever cropped up in the future.

```
    return new Reflections(Resettable.getPackage().getName()).getSubTypesOf( Resettable.class ).findAll { !Traits.isTrait(it) }
```

`findAll` should select all the elements that are NOT a trait (and the only trait is Pluggable at the moment), and we're good (because all subtypes of Pluggable are already found by `getSubTypesOf(Resettable.class)`.